var/spack/repos/builtin/packages/mofem-fracture-module/package.py

```diff
@@ -0,0 +1,97 @@
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+
+
+class MofemFractureModule(CMakePackage):
+    """mofem fracture module"""
+
+    homepage = "http://mofem.eng.gla.ac.uk"
+    url = "https://bitbucket.org/likask/mofem_um_fracture_mechanics"
+
+    maintainers = ['likask']
+
+    version('0.9.42',
+        git='https://bitbucket.org/likask/mofem_um_fracture_mechanics',
+        tag='v0.9.42')
+    version('develop',
+        git='https://bitbucket.org/likask/mofem_um_fracture_mechanics',
+        branch='develop')
+
+    variant('copy_user_modules', default=True,
+        description='Copy user modules directory instead linking')
+
+    extends('mofem-cephas')
+    depends_on("mofem-users-modules", type=('build', 'link', 'run'))
+
+    @property
+    def root_cmakelists_dir(self):
+        """The relative path to the directory containing CMakeLists.txt
+
+        This path is relative to the root of the extracted tarball,
+        not to the ``build_directory``. Defaults to the current directory.
+
+        :return: directory containing CMakeLists.txt
+        """
+        spec = self.spec
+        return os.path.join(spec['mofem-cephas'].prefix.users_modules)
+
+    def cmake_args(self):
+        spec = self.spec
+        source = self.stage.source_path
+
+        options = []
+
+        # obligatory options
+        options.extend([
+            '-DWITH_SPACK=YES',
+            '-DEXTERNAL_MODULES_BUILD=YES',
+            '-DUM_INSTALL_BREFIX=%s' % spec['mofem-users-modules'].prefix,
+            '-DEXTERNAL_MODULE_SOURCE_DIRS=%s' % source,
+            '-DSTAND_ALLONE_USERS_MODULES=%s' %
+            ('YES' if '+copy_user_modules' in spec else 'NO')])
+
+        # Set module version
+        if self.spec.version[0] == 'develop':
+            options.extend([
+                '-DFM_VERSION_MAJOR=%s' % 0,
+                '-DFM_VERSION_MINOR=%s' % 0,
+                '-DFM_VERSION_BUILD=%s' % 0])
+        else:
+            options.extend([
+                '-DFM_VERSION_MAJOR=%s' % self.spec.version[0],
+                '-DFM_VERSION_MINOR=%s' % self.spec.version[1],
+                '-DFM_VERSION_BUILD=%s' % self.spec.version[2]])
+
+        # build tests
+        options.append('-DMOFEM_UM_BUILD_TETS={0}'.format(
+            'ON' if self.run_tests else 'OFF'))
+
+        return options
+
+    @run_after('install')
```

Comment:
Out of curiosity, what exactly does the `make install` stage do?

Reply:
Install process is standard, cmake, build, install.

@run_after('install') is not needed to run code installed by extension, nor in the install process. However for mofem tutorial, for example, mofem-minimal-equations, users like to have access to source code to play with it. That why I copy it to install prefix.  

Having source code at hand one can compile in own build directory it in mofem-cephas view when the extension is activated.

Reply:
i would add that clarification here as well. In half a year we all will forge where that PR is or discussion we had here, so it's good to have a quick hints in the package file itself. Especially given that this is not a typical use-case.